src/store/user/user-context.js

```diff
@@ -10,6 +11,27 @@ export const UserContextProvider = ({ children }) => {
   const [showMemberRoleUpdateModal, setShowMemberRoleUpdateModal] =
     useState(false);
   const [selectedMember, setSelectedMember] = useState(null);
+  const [superUserVerified, setSuperUserVerified] = useState(false);
+
+  const verifySuperUser = async () => {
+    let selfUser = null;
+    let selfIsSuperUser = false;
+    let selfSuperUserVerified = false;
+    try {
+      const { data } = await getUserSelf();
+      if (data) {
+        selfUser = data;
+        selfIsSuperUser = Boolean(data.roles?.super_user);
+        selfSuperUserVerified = true;
```

Comment:
Why is this needed? Why can't we just reuse `selfIsSuperUser`?

Reply:
probably the name 'selfSuperUserVerified`confused you and vinayak-g. I have changed it to `selfUserApiCalled`. This variable prevents us from making multiple API calls.